clash-lib/src/Clash/Core/TermLiteral.hs

```diff
@@ -80,6 +82,14 @@ instance TermLiteral Natural where
     Right (fromInteger n)
   termToData t = Left t
 
+newtype TermLiteralSNat = TermLiteralSNat Natural
+  deriving (Show)
+
+instance TermLiteral TermLiteralSNat where
```

Comment:
So turns out I misinterpreted why we couldn't make an instance for `SNat`: this class's superclass is `Typeable`, which needs a `KnownNat` constraint. We only use it to pretty print types, so we could make a specialized function on this class for that, defaulting to the `Typeable` stuff but just returning `"SNat"` for that instance.

Reply:
This has been fixed in https://github.com/clash-lang/clash-compiler/pull/2329, so I'll rebase this PR and change the API.

Reply:
This is done, right?

Reply:
Yup